doc/source/whatsnew/v1.2.0.rst

```diff
@@ -621,6 +621,7 @@ Indexing
 - Bug in :meth:`DataFrame.iloc` and :meth:`Series.iloc` aligning objects in ``__setitem__`` (:issue:`22046`)
 - Bug in :meth:`DataFrame.loc` did not raise ``KeyError`` when missing combination was given with ``slice(None)`` for remaining levels (:issue:`19556`)
 - Bug in :meth:`DataFrame.loc` raising ``TypeError`` when non-integer slice was given to select values from :class:`MultiIndex` (:issue:`25165`, :issue:`24263`)
+- Bug in :meth:`DataFrame.loc` returning and assigning elements in wrong order when indexer is differently ordered than the :class:`MultiIndex` to filter (:issue:`31330`, :issue:`34603`)
```

Comment:
im assuming the merge conflict is here.  sometimes ill add whatsnew notes a few lines up to avoid merge conflicts

Reply:
Ah, I somehow though they should be chronological, will do this in the future